feat: produce OpenTelemetry traces with `hs-opentelemetry` (#3140)
PostgREST/postgrest: docs/references/observability.rst

  HTTP/1.1 200 OK
  X-Request-Id: 123

.. _opentelemetry:

OpenTelemetry
-------------

PostgREST is able to produce OpenTelemetry traces. OpenTelemetry is configured
using ``OTEL_*`` environment variables, per the `OpenTelemetry specification`_.

The OpenTelemetry support is currently both experimental and in early stages of development, so expect some rough edges

> or lack of functionality, such as metrics or logs. Since current OpenTelemetry implementation incurs a small
> (~6% in our "Loadtest (mixed)" suite) performance hit, it is gated behind the :ref:`server-otel-enabled`

Reviewer: ```suggestion
or lack of functionality, such as metrics or logs. Since the current OpenTelemetry implementation incurs a small performance penalty, it is gated behind the :ref:`server-otel-enabled`
```

I don't think we need to reference our loadtest here.

configuration option, disabled by default.

Example configuration:

.. code-block:: shell

    OTEL_EXPORTER_OTLP_ENDPOINT='https://api.honeycomb.io/' \
    OTEL_EXPORTER_OTLP_HEADERS="x-honeycomb-team=<honeycomb_api_key>"  \
    OTEL_SERVICE_NAME='PostgREST'\
    OTEL_LOG_LEVEL='debug'\
    OTEL_TRACES_SAMPLER='always_on' \
    postgrest

Prometheus metrics through the OpenTelemetry Collector
------------------------------------------------------

> PostgREST currently exports traces through OpenTelemetry, but not metrics. But,
> it's possible to scrape the :ref:`metrics` endpoint exposed by the
> :ref:`admin_server` and relay it through the OpenTelemetry Collector.

Reviewer: ```suggestion
PostgREST currently only exports traces through OpenTelemetry. To get metrics,
it's possible to scrape the :ref:`metrics` endpoint exposed by the
:ref:`admin_server` and relay it through the OpenTelemetry Collector.
```
